actors/builtin/miner/miner_state_test.go

```diff
@@ -0,0 +1,587 @@
+		assert.Equal(t, append(sectorExpirations[exp1], mergedSectors...), merged)
+	})
+
+	t.Run("clear sectors by expirations", func(t *testing.T) {
+		rt := builder.Build(t)
+		harness := constructStateHarness(t, rt, abi.ChainEpoch(0))
+
+		harness.assertAddSectorExpiration(rt, exp1, sectorExpirations[exp1]...)
+		harness.assertAddSectorExpiration(rt, exp2, sectorExpirations[exp2]...)
+
+		// ensure clearing works
+		harness.assertClearSectorExpiration(rt, exp1, exp2)
+		empty1 := harness.assertGetSectorExpirations(rt, exp1)
+		assert.Empty(t, empty1)
+
+		empty2 := harness.assertGetSectorExpirations(rt, exp2)
+		assert.Empty(t, empty2)
+	})
+
+	t.Run("fail to add more than SectorsMax number of sectors", func(t *testing.T) {
+		rt := builder.Build(t)
+		harness := constructStateHarness(t, rt, abi.ChainEpoch(0))
+
+		tooManySectors := make([]uint64, miner.SectorsMax+1)
+
+		for i := uint64(0); i < miner.SectorsMax+1; i++ {
+			tooManySectors[i] = i
+		}
+
+		err := harness.s.AddSectorExpirations(adt.AsStore(rt), 1, tooManySectors...)
+		assert.Error(t, err)
+
+	})
+
+}
+
+type minerStateHarness struct {
+	s *miner.State
+	t testing.TB
+
+	cidGetter func() cid.Cid
+	seed      uint64
+}
+
+//
+// Sector Expiration Store
+//
+
+// internall converst the bit field to slice of uint64
+func (h *minerStateHarness) assertGetSectorExpirations(rt runtime.Runtime, expiry abi.ChainEpoch) []uint64 {
+	bf, err := h.s.GetSectorExpirations(adt.AsStore(rt), expiry)
+	assert.NoError(h.t, err)
+	sectors, err := bf.All(miner.SectorsMax)
+	assert.NoError(h.t, err)
+	return sectors
+}
+
+func (h *minerStateHarness) assertAddSectorExpiration(rt runtime.Runtime, expiry abi.ChainEpoch, sectors ...uint64) {
+	err := h.s.AddSectorExpirations(adt.AsStore(rt), expiry, sectors...)
+	assert.NoError(h.t, err)
+}
+
+func (h *minerStateHarness) assertRemoveSectorExpiration(rt runtime.Runtime, expiry abi.ChainEpoch, sectors ...uint64) {
+	err := h.s.RemoveSectorExpirations(adt.AsStore(rt), expiry, sectors...)
+	assert.NoError(h.t, err)
+}
+
+func (h *minerStateHarness) assertClearSectorExpiration(rt runtime.Runtime, excitations ...abi.ChainEpoch) {
+	err := h.s.ClearSectorExpirations(adt.AsStore(rt), excitations...)
+	assert.NoError(h.t, err)
+}
+
+//
+// NewSectors BitField Assertions
+//
+
+func (h *minerStateHarness) assertAddNewSectors(sectorNos ...abi.SectorNumber) {
+	err := h.s.AddNewSectors(sectorNos...)
+	assert.NoError(h.t, err)
+}
+
+// makes a bit field from the passed sector numbers
+func (h *minerStateHarness) assertRemoveNewSectors(sectorNos ...abi.SectorNumber) {
+	bf := newBitFieldFromSectorNos(sectorNos...)
+	err := h.s.RemoveNewSectors(bf)
+	assert.NoError(h.t, err)
+}
+
+func (h *minerStateHarness) assertClearNewSectors() {
+	h.assertNewSectorsCount(0)
+}
+
+func (h *minerStateHarness) assertNewSectorsCount(count uint64) {
+	sectorCount, err := h.s.NewSectors.Count()
+	assert.NoError(h.t, err)
+	assert.Equal(h.t, count, sectorCount)
```

Comment:
In general I think pushing the semantically relevant assertions into helpers like this detracts from test clarity. This would be better as a `countNewSectors()` method that returns the count (but no error), and then putting the `assert.Equal` in the calling test code directly.

So the harness generally would have a bunch of `require.NoError` but no/very few `assertEqual`, and thus ~none of the method names start with `assert`. There may be some exceptions for a complex but repeatedly-used assertion.